zanata-war/src/main/java/org/zanata/adapter/OkapiFilterAdapter.java

```diff
@@ -172,6 +168,9 @@ public Resource parseDocumentFile(URI documentContent,
         RawDocument rawDoc =
                 new RawDocument(documentContent, "UTF-8",
                         net.sf.okapi.common.LocaleId.fromString("en"));
+        if (rawDoc.getTargetLocale() == null) {
```

Comment:
You *just* created the document.  Shouldn't you know whether its target locale is set?